src/components/TypeAhead.js

```diff
@@ -86,7 +86,9 @@ const TypeAhead = React.createClass({
     //add to selectedItems
     const selectedItems = this.state.selectedItems;
 
-    selectedItems.push(item);
+    if (item) {
+      selectedItems.push(item);
+    }
```

Comment:
Since this bug is only happening on tab, it would be better to check for the existence of `item` before we even call `_handleItemSelect`. On line 133 we are assuming that `filteredItems` has at least one value when that isn't always the case.